src/core/simulation_results.jl

```diff
@@ -1,616 +1,133 @@
-    stage = "stage-$stage_name"
-    references = sim_output.ref
-    base_power = sim_output.base_powers[stage_name]
-    variables = Dict{Symbol, DataFrames.DataFrame}()
-    duals = Dict{Symbol, Any}()
-    params = Dict{Symbol, DataFrames.DataFrame}()
-    time_stamp = DataFrames.DataFrame(Range = Dates.DateTime[])
-    time_length = sim_output.chronologies[stage]
-    dual = _find_duals(collect(keys(references[stage])))
-    param = _find_params(collect(keys(references[stage])))
-    variable = setdiff(variable, vcat(param, dual))
-    for l in 1:length(variable)
-        date_df = references[stage][variable[l]]
-        step_df = DataFrames.DataFrame(
-            Date = Dates.DateTime[],
-            Step = String[],
-            File_Path = String[],
-        )
-        for n in 1:length(step)
-            step_df = vcat(step_df, date_df[date_df.Step .== step[n], :])
-        end
-        variables[(variable[l])] = DataFrames.DataFrame()
-        for (ix, time) in enumerate(step_df.Date)
-            file_path = step_df[ix, :File_Path]
-            var = read_arrow_file("$file_path")
-            variables[(variable[l])] = vcat(variables[(variable[l])], var[1:time_length, :])
-            if l == 1
-                time_stamp = vcat(time_stamp, _read_time(file_path, time_length))
-                check_file_integrity(dirname(file_path))
-            end
-        end
+function results_pre_process(store::HdfSimulationStore, stage_name::Symbol)
+    stage_results_params = Dict()
+    store_params = get_params(store)
+    #sim_initial_time = get_initial_time(store_params)
+    stages_params = get_stages(store_params)
+    if stage_name ∉ keys(stages_params)
+        error("Stage with name $(stage_name) not in the simulation")
     end
-    time_stamp[!, :Range] = convert(Array{Dates.DateTime}, time_stamp[!, :Range])
-    file_path = dirname(references[stage][variable[1]][1, :File_Path])
-    optimizer = read_json(joinpath(file_path, "optimizer_log.json"))
-    obj_value = Dict{Symbol, Any}(:OBJECTIVE_FUNCTION => optimizer["obj_value"])
-    duals = _read_references(duals, dual, stage, step, references, time_length)
-    param_values = _read_references(params, param, stage, step, references, time_length)
-    return SimulationResults(
-        base_power,
-        variables,
-        obj_value,
-        optimizer,
-        time_stamp,
-        duals,
-        results_folder,
-        param_values,
-    )
+    stage_params = stages_params[stage_name]
+    stage_results_params["base_power"] = get_base_power(stage_params)
+    stage_results_params["system_uuid"] = get_system_uuid(stage_params)
+    stage_dataset = get_dataset(store, stage_name)
+    stage_results_params["available_variables"] = keys(get_variables(stage_dataset))
+    stage_results_params["available_params"] = keys(get_parameters(stage_dataset))
+    stage_results_params["available_duals"] = keys(get_duals(stage_dataset))
```

Comment:
We need a better interface for this. I expected that we _might_ need something like this and left this comment in the HDF file:
```
# TODO: Interfaces to add
# - list_stages(store)
# - list_names(store, type) for duals/parameters/variables by stage
# - get_step_resolution(store)
# - get_resolution(store, stage)
```
What you think about those names?

Reply:
What is type in `# - list_names(store, type) for duals/parameters/variables by stage` ? 

Otherwise, the names look good to me.

Reply:
It should be one of these constants: https://github.com/NREL-SIIP/PowerSimulations.jl/blob/2bbbc790c1a425c4fe792ef697e8be81e4bf8d75/src/core/simulation_store.jl#L1

Reply:
I would prefer not to use the word type here since it can be confusing. I thought that these will be structs

Reply:
The SimulationStore functions `read_result`, `write_result`, and `list_names` need to be passed arguments specifying dual/parameter/variable and the symbol name. Let me know your preference.

Reply:
value? 

```Julia
const CONTAINER_DUALs_VALUE = :duals
const CONTAINER_PARAMETERS_VALUE = :parameters
const CONTAINER_VARIABLES_VALUE = :variables
const CONTAINER_VALUE =
    Set((CONTAINER_DUALs_VALUE,CONTAINER_PARAMETERS_VALUE, CONTAINER_VARIABLES_VALUE))
```

Reply:
CONTAINER_DUALS, CONTAINER_PARAMETERS, CONTAINER_VARIABLES

Reply:
Sounds good